### PR TITLE
feature: `BlankLineBeforeStatementFixer` - skip enum cases

### LINKED ISSUE #6747: blank_line_before_statement: prevent adding line before enum case
## Feature request
Given:
```php
enum Foo
{
    case One;
    case Two;
}
```

`blank_line_before_statement`, with `case` configured will transform to:

```php
enum Foo
{
    case One;
    
    case Two;
}
```

This is undesirable formatting.

An option could be introduced to somehow exclude these specific enum`cases`.

### PR DESCRIPTION
**Pull request by paulbalandan**
Alternative to #7201 
Closes #6747

### CODE REVIEW

File: src/Tokenizer/TokensAnalyzer.php [hunk 1]
```diff
@@ -633,6 +633,28 @@ public function isWhilePartOfDoWhile(int $index): bool
...
+            ));
+        }
+
+        if (!\defined('T_ENUM') || !$tokens->isTokenKindFound(T_ENUM)) {
+            return false;
+        }
+
+        $prevIndex = $tokens->getPrevTokenOfKind($caseIndex, [[T_ENUM], [T_SWITCH]]);
+
+        return null !== $prevIndex && $tokens[$prevIndex]->isGivenKind(T_ENUM);
```
Comment (Wirone) [inline src/Tokenizer/TokensAnalyzer.php:655]: I think that with this implementation non-enum cases inside enum (like in `pure enum with switch` test case) won't be considered as statements too, so they won't be fixed with `--rules='{"blank_line_before_statement":{"statements":["case"]}}'` configuration.

Comment (paulbalandan) [reply src/Tokenizer/TokensAnalyzer.php:655]: switch cases inside enums will still be fixed. Look at the modified provideFix81Cases of BlankLineBeforeStatementFixerTest

Comment (Wirone) [reply src/Tokenizer/TokensAnalyzer.php:655]: OK, I tested it wrong. I took `enum Suit` from and applied fixer, but it has two cases with same logic and new line is not applied then. When cases have different logic, new line is added:

image

And yeah, now I get the logic here - find `enum` or `switch`, but only on of them makes it an "enum case" 👍.

File: src/Tokenizer/TokensAnalyzer.php [hunk 2]
```diff
@@ -633,6 +633,28 @@ public function isWhilePartOfDoWhile(int $index): bool
         return $tokens[$beforeStartIndex]->isGivenKind(T_DO);
     }
 
+    public function isEnumCase(int $caseIndex): bool
```
Comment (Wirone) [inline src/Tokenizer/TokensAnalyzer.php:636]: Non-blocking, but consider adding proper `@throws LogicException when provided index does not point to token containing T_CASE`.

Comment (paulbalandan) [reply src/Tokenizer/TokensAnalyzer.php:639]: Added

**Review by kubawerlos — APPROVED** [review]

**Comment by Wirone** [comment]
Thank you @paulbalandan 🍻

**Pull request merged**